src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,114 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private T[] innerArray = (T[]) new Object[DEFAULT_CAPACITY];
```

Comment:
The unchecked cast `(T[]) new Object[DEFAULT_CAPACITY]` can lead to a `ClassCastException` at runtime. Consider using `@SuppressWarnings("unchecked")` to suppress the warning if you are sure about the type safety.